### ISSUE TITLE
DatePicker: focusTrap doesn't work in Safari

### ISSUE THREAD

**Issue by nikitaorliak-cengage** [cengage/react-magma#1910]
**Describe the bug**
focusTrap doesn't work in Safari. When you press the **Tab** key on the last element inside the DatePicker, it exits it.

**To Reproduce**
1. Open MacOS + Safari.
2. Open the documentation.
3. Navigate to DatePicker.
4. Focus on the input.
5. Press **Tab**.
6. Verify that the focus has moved to another DatePicker. (broken behavior №1)
7. Open DatePicker.
8. Focus on the date.
9. Press **Tab**.
10. Verify that the focus has left the DatePicker. (broken behavior №2)

**Expected behavior**
Focus locks inside the DatePicker.

**Notes**
Test DateTimePicker also
